Assets/MRTK/SDK/Editor/Inspectors/UX/InteractiveElement/TrackedStatesInspector.cs

```diff
@@ -0,0 +1,282 @@
+                                {
+                                    newStateName = EditorGUILayout.TextField("New State Name", newStateName);
+
+                                    if (GUILayout.Button("Set State Name"))
+                                    {
+                                        stateName.stringValue = newStateName;
+                                    }
+                                }
+                            }
+                        }
+                    }
+
+                    EditorGUILayout.Space();                    
+                }
+            }
+        }
+
+        // Render the event configuration of a state
+        private void RenderStateEventConfiguration(SerializedProperty stateName, SerializedProperty stateEventConfiguration)
+        {
+            // If the current state does have a matching event configuration but the event configuration is null, draw a button to 
+            // give the user an option to 'Add State Events'.
+            if (stateEventConfiguration.objectReferenceValue == null && StateHasExistingEventConfiguration(stateName.stringValue))
+            {
+                // Draw a button in the state container titled 'Add State Event'. On button press, an instance of the matching
+                // event configuration scriptable is created and assigned to the state
+                if (GUILayout.Button("Add " + stateName.stringValue + " State Events"))
+                {
+                    CreateEventScriptable(stateEventConfiguration, stateName.stringValue);
+                }
+            }
+            else if (stateEventConfiguration.objectReferenceValue != null)
+            {
+                string stateFoldoutID = stateName.stringValue + "EventConfiguration";
+
+                // Draw a foldout for the event configuration
+                if (InspectorUIUtility.DrawSectionFoldoutWithKey(stateName.stringValue + " State Events", stateFoldoutID, MixedRealityStylesUtility.TitleFoldoutStyle, false))
+                {
+                    // Draw the events for the associated state if this state has an event configuration
+                    DrawStateEventsScriptableSubEditor(stateEventConfiguration);
+                }
+            }
+        }
+
+        // Check if the given state has a valid event configuration to render
+        private bool StateHasExistingEventConfiguration(string stateName)
+        {
+            // Get the list of the subclasses for BaseInteractionEventConfiguration and find the
+            // event configuration that contains the given state name
+            var eventConfigurationTypes = TypeCacheUtility.GetSubClasses<BaseInteractionEventConfiguration>();
+            var eventConfigType = eventConfigurationTypes.Find(t => t.Name.StartsWith(stateName));
+
+            // If there is associated event configuration for the state, then it is valid
+            if (eventConfigType != null)
```

Comment:
You can use [boolean zen](https://courses.cs.washington.edu/courses/cse143/16au/style/bad-boolean-zen.html) to shorten this expression to: `return eventConfigurationTypes.Find(t => t.Name.StartsWith(stateName)) != null`, or better yet, using [Contains](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.contains?view=netcore-3.1) method.

Reply:
The contains method was used initially but if a developer creates a state and decides to name it "e" or any other single character,  the contains method would return true for any event configuration name that contains the letter "e".

Reply:
Not String.Contains, the [Enumerable.Contains](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.contains?view=netcore-3.1) :-)

So the way I would write that might be:

```
return eventConfigurationTypes.Contains(t => t.Name.StartsWith(stateName));
```

NOTE: after reading Contains documentation, it looks like contains doesn't take functions, so I would instead write it like this:

```
return eventConfigurationTypes.Find(t => t.Name.StartsWith(stateName)) != null;
```

The more important lesson is to try and avoid code like this:

```
if (variable)
{
    return true;
}
else
{
    return false;
}
```

and instead write:

```
return variable;
```

Reply:
Already updated :)